Source/SIMPLib/REST/V1Controllers/SIMPLStaticFileController.cpp

```diff
@@ -0,0 +1,228 @@
+
+#include "SIMPLStaticFileController.h"
+#include <QtCore/QDateTime>
+#include <QtCore/QDir>
+#include <QtCore/QFileInfo>
+
+#include "SIMPLib/Plugin/SIMPLPluginConstants.h"
+
+#include "SIMPLib/REST/SIMPLDirectoryListing.h"
+
+SIMPLStaticFileController* SIMPLStaticFileController::m_Instance = nullptr;
+
+// -----------------------------------------------------------------------------
+//
+// -----------------------------------------------------------------------------
+SIMPLStaticFileController* SIMPLStaticFileController::Instance()
+{
+  return m_Instance;
+}
+
+// -----------------------------------------------------------------------------
+//
+// -----------------------------------------------------------------------------
+void SIMPLStaticFileController::CreateInstance(ServerSettings* settings, QObject* parent)
+{
+  if(m_Instance != nullptr)
+  {
+    delete m_Instance;
```

Comment:
`delete nullptr;` is valid C++.  You do not need the nullptr check.

Reply:
Really?  Did not know that.  Will keep that in mind in the future.  I assumed `delete nullptr;` would result in a crash.

Reply:
Before C++14, no destructors were called on nullptr, but it was implementation defined if the deallocation functions were called (which were already guaranteed to do nothing when given null pointers).  C++14 simplified this to never call deallocation functions on null pointers.

Reply:
Done